### PR TITLE
Add more sourceforge mirrors for D-APT to base image

### PR DESCRIPTION
**Pull request by nemanja-boric-sociomantic**

### CODE REVIEW

File: docker/base [hunk 1]
```diff
@@ -41,6 +41,11 @@ apt-key adv --keyserver keyserver.ubuntu.com --recv-keys EBCF975E5BA24D5E
 wget http://master.dl.sourceforge.net/project/d-apt/files/d-apt.list \
 		-O /etc/apt/sources.list.d/d-apt.list
 
+echo "deb http://freefr.dl.sourceforge.net/project/d-apt/ d-apt main" >> \
```
Comment (leandro-lucarella-sociomantic) [inline docker/base]: Maybe add a comment saying we are adding a few backup mirrors?

Also, maybe move the `>>` to the next line to match the current style?

**Review by leandro-lucarella-sociomantic — COMMENTED** [review]
Not sure if we should just put this in v4.x.x, this code was moved to `docker/dlang` and it would be a way to *encourage* people to move to v4... :D

File: docker/base (hunk 1 above)
Comment (nemanja-boric-sociomantic) [reply docker/base]: :+1:

**Comment by leandro-lucarella-sociomantic** [comment]
Also, if this is targeting a minor branch, it would be good to include a release note.

**Comment by leandro-lucarella-sociomantic** [comment]
Aaaand it fails. :cry:

**Comment by leandro-lucarella-sociomantic** [comment]
```
E: Failed to fetch http://master.dl.sourceforge.net/project/d-apt/dists/d-apt/InRelease  Clearsigned file isn't valid, got 'NOSPLIT' (does the network require authentication?)
```

Tricky. We could ignore the results of `apt-get update`, but it might be dodgy.

**Comment by leandro-lucarella-sociomantic** [comment]
Maybe doing a specific hack for SF, do a ping to the host or something, but in this case is just `apt` that's failing, right? Is not that the host is down?

**Comment by nemanja-boric-sociomantic** [comment]
That's correct - the host is still online.

**Comment by nemanja-boric-sociomantic** [comment]
At least rebased with to the v4.x.x, fixed the formatting and added the release note.

**Comment by leandro-lucarella-sociomantic** [comment]
I hijacked your PR to add a dumb heuristic to retry new mirrors if `apt update` fails.

**Comment by leandro-lucarella-sociomantic** [comment]
```
+ apt-key adv --keyserver keyserver.ubuntu.com --recv-keys EBCF975E5BA24D5E
Executing: /tmp/tmp.lhnY8UxVpM/gpg.1.sh --keyserver
keyserver.ubuntu.com
--recv-keys
EBCF975E5BA24D5E
gpg: requesting key 5BA24D5E from hkp server keyserver.ubuntu.com
gpg: no valid OpenPGP data found.
gpg: Total number processed: 0
gpg: keyserver communications error: keyserver helper general error
gpg: keyserver communications error: unknown pubkey algorithm
gpg: keyserver receive failed: unknown pubkey algorithm
gpgkeys: key EBCF975E5BA24D5E can't be retrieved
```

WTF now.

**Comment by leandro-lucarella-sociomantic** [comment]
For what I've read (very little info about it) it seems like it could be also a network problem with the key server, but it does work locally.

**Comment by leandro-lucarella-sociomantic** [comment]
I'll re-run to see if it's a transient problem.

**Comment by leandro-lucarella-sociomantic** [comment]
:green_heart: :disappointed_relieved:

**Comment by leandro-lucarella-sociomantic** [comment]
Working as expected!

* https://travis-ci.org/sociomantic-tsunami/cachalot/builds/380307718#L1832
* https://travis-ci.org/sociomantic-tsunami/cachalot/builds/380307718#L1837

**Comment by leandro-lucarella-sociomantic** [comment]
Wait, no, this is shit too. This trick only works when building the image, but then when it is used, it will just use the mirror that worked at the time to build the image.

I think there is no (reasonable) way around it that is not using a more reliable provider...

**Comment by mihails-strasuns-sociomantic** [comment]
I actually think ignoring results of `apt update` is least dodgy option. If it went really wrong, our `apt install` will fail on next line anyway.

**Comment by nemanja-boric-sociomantic** [comment]
I agree with @mihails-strasuns here.

**Comment by leandro-lucarella-sociomantic** [comment]
> I actually think ignoring results of apt update is least dodgy option. If it went really wrong, our apt install will fail on next line anyway.

But this approach will end up in, either failing anyway or if it works, you might end up testing (and deploying) with an old version without even noticing. I'm not sure that's a good idea.

**Comment by nemanja-boric-sociomantic** [comment]
Alternative idea suggested by Seb is to use dlang install script: https://github.com/dlang/ci/issues/210#issuecomment-390160673

**Comment by leandro-lucarella-sociomantic** [comment]
I would prefer to avoid going outside of Debian packages.

**Comment by leandro-lucarella-sociomantic** [comment]
We could just try to build a SF mirror somewhere else. After all is just some HTTP mirror, it shouldn't be that hard.

**Comment by leandro-lucarella-sociomantic** [comment]
I'll close as wontfix. I don't think this is the right approach.

**Pull request closed without merging**